crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -311,10 +312,20 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
                     (Some((multiple_capture_data, capture_update_list)), None)
                 }
                 None => {
-                    let status = match err.status_code {
-                        500..=511 => storage::enums::AttemptStatus::Pending,
-                        _ => storage::enums::AttemptStatus::Failure,
-                    };
+                    let flow_name = core_utils::get_flow_name::<F>()?;
+                    let status =
+                        // mark previous attempt status for technical failures in PSync flow
+                        if flow_name == "PSync" {
+                            match err.status_code {
+                                200..=299 => storage::enums::AttemptStatus::Failure,
```

Comment:
Why 2xx are marked as failures?

Reply:
If we get 2xx response from connector, and connector flows sends us the error response then it's a payment failure right? If we don't do this we would never be marking payment to terminal state in psync flows